src/python/dxpy/cli/exec_io.py

```diff
@@ -431,18 +436,33 @@ def __init__(self, executable=None, input_name_prefix=None, input_spec=None):
             inaccessible_stages = [stage['id'] for stage in self._desc['stages'] if stage['accessible'] is False]
             raise DXCLIError('The workflow ' + self._desc['id'] + ' has the following inaccessible stage(s): ' + ', '.join(inaccessible_stages))
 
+        # Workflow-level inputs (defined in workflowInputSpec)
+        #  i. The workflow has no workflowInputSpec
+        #   * The inputs can be passed to stages directly
+        # ii. The workflow has workflowInputSpec (in a closed or open state)
```

Comment:
This makes sense for consistent UX for applets and workflows.  This note is a reminder for us to discuss the implications of this in subsequent design reviews.